### PR TITLE
Game Fragment Functionality Added 

### PR DESCRIPTION
**Pull request by akshat-sahijpal**

### CODE REVIEW

File: app/src/main/java/com/pallaw/swipeandlearnf/feature/data/CardQuestionData.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.pallaw.swipeandlearnf.feature.data
+
+data class CardQuestionData(
```
Comment (pallaw) [inline app/src/main/java/com/pallaw/swipeandlearnf/feature/data/CardQuestionData.kt]: @akshat-sahijpal use existing Question data class in domain layer

File: app/src/main/java/com/pallaw/swipeandlearnf/feature/adapter/QuestionsAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import android.widget.TextView
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import com.pallaw.swipeandlearnf.R
+import com.pallaw.swipeandlearnf.feature.data.CardQuestionData
+import com.yuyakaido.android.cardstackview.CardStackListener
+import com.yuyakaido.android.cardstackview.Direction
+
+class QuestionsAdapter(
+    private val questionsData: List<CardQuestionData>,
```
Comment (pallaw) [inline app/src/main/java/com/pallaw/swipeandlearnf/feature/adapter/QuestionsAdapter.kt]: @akshat-sahijpal use existing Question data class in domain layer

File: app/src/main/java/com/pallaw/swipeandlearnf/feature/adapter/QuestionsAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import android.widget.RelativeLayout
+import android.widget.TextView
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import com.pallaw.swipeandlearnf.R
+import com.pallaw.swipeandlearnf.feature.data.CardQuestionData
+import com.yuyakaido.android.cardstackview.CardStackListener
+import com.yuyakaido.android.cardstackview.Direction
+
+class QuestionsAdapter(
```
Comment (pallaw) [inline app/src/main/java/com/pallaw/swipeandlearnf/feature/adapter/QuestionsAdapter.kt]: @akshat-sahijpal move this class to Ui/game folder

**Review by pallaw — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pallaw/swipeandlearnf/feature/adapter/QuestionsAdapter.kt (hunk 3 above)
Comment (akshat-sahijpal) [reply app/src/main/java/com/pallaw/swipeandlearnf/feature/adapter/QuestionsAdapter.kt]: done

File: app/src/main/java/com/pallaw/swipeandlearnf/feature/data/CardQuestionData.kt (hunk 1 above)
Comment (pallaw) [reply app/src/main/java/com/pallaw/swipeandlearnf/feature/data/CardQuestionData.kt]: @akshat-sahijpal delete this class if it's not being used anywhere

File: app/src/main/java/com/pallaw/swipeandlearnf/domain/model/Question.kt [hunk 4]
```diff
@@ -1,7 +1,7 @@
 package com.pallaw.swipeandlearnf.domain.model
 
 data class Question(
-    val answer: Boolean,
-    val hint: String,
-    val question: String
+    val answer: Boolean?=null,
```
Comment (pallaw) [inline app/src/main/java/com/pallaw/swipeandlearnf/domain/model/Question.kt]: @akshat-sahijpal why nullable?

**Review by pallaw — CHANGES_REQUESTED** [review]

**Review by pallaw — APPROVED** [review]

**Pull request merged**